### PR TITLE
P1 tests added for PostFX Layer, PostFX Gradient and PostFX Shape

### PR DESCRIPTION
**Pull request by smurly**
Signed-off-by: Scott Murray <scottmur@amazon.com>

## What does this PR do?
P1 test automation for
- PostFX Layer component
- PostFX Shape Weight Modifier component
- PostFX Gradient Weight Modifier component
an issue was encountered trying to set camera tag values with PostFX Layer component and a new GHI was created for follow-up. The steps that did not work are commented out pending further investigation. Manually setting camera tags was functioning as expected.

## How was this PR tested?
locally in editor and at command line
```
Report:
[SUCCESS] Success: PostFX Layer Entity successfully created
[SUCCESS] Success: Entity has a PostFX Layer component
[SUCCESS] Success: Remove PostFX Layer component success
[SUCCESS] Success: Entity has a PostFX Layer component
[SUCCESS] Success: 'Layer Category' set to FrontEnd
[SUCCESS] Success: 'Layer Category' set to Cinematics
[SUCCESS] Success: 'Layer Category' set to Gameplay
[SUCCESS] Success: 'Layer Category' set to Camera
[SUCCESS] Success: 'Layer Category' set to Volume
[SUCCESS] Success: 'Layer Category' set to Level
[SUCCESS] Success: 'Layer Category' set to Default
[SUCCESS] Success: 'Priority' property set
[SUCCESS] Success: 'Weight' property set
[SUCCESS] Success: 'Select Camera Tags Only' is a container property
[SUCCESS] Success: 'Excluded Camera Tags' is a container property
[SUCCESS] Success: Entered game mode
[SUCCESS] Success: Exited game mode
[SUCCESS] Success: Entity is hidden
[SUCCESS] Success: P0: Entity is visible
[SUCCESS] Success: Entity deleted
[SUCCESS] Success: UNDO deletion success
[SUCCESS] Success: REDO deletion success
Test result:  SUCCESS
```
```
Report:
[SUCCESS] Success: PostFx Shape Weight Modifier Entity successfully created
[SUCCESS] Success: Entity has a PostFx Shape Weight Modifier component
[SUCCESS] Success: PostFx Shape Weight Modifier component removed
[SUCCESS] Success: Entity has a PostFx Shape Weight Modifier component
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a PostFX Layer component
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Axis Aligned Box Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Box Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Capsule Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Compound Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Cylinder Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Disk Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Polygon Prism Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Quad Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Sphere Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Shape Reference component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: PostFx Shape Weight Modifier component disabled
[SUCCESS] Success: Entity has a Tube Shape component
[SUCCESS] Success: PostFx Shape Weight Modifier component enabled
[SUCCESS] Success: 'Fall-off Distance' property set
[SUCCESS] Success: Entered game mode
[SUCCESS] Success: Exited game mode
[SUCCESS] Success: Entity is hidden
[SUCCESS] Success: Entity is visible
[SUCCESS] Success: Entity deleted
[SUCCESS] Success: UNDO deletion success
[SUCCESS] Success: REDO deletion success
Test result:  SUCCESS
```
```
Report:
[SUCCESS] Success: PostFX Gradient Weight Modifier Entity successfully created
[SUCCESS] Success: Entity has a PostFX Gradient Weight Modifier component
[SUCCESS] Success: Entity has a PostFX Gradient Weight Modifier component
[SUCCESS] Success: Entity has a PostFX Gradient Weight Modifier component
[SUCCESS] Success: PostFX Gradient Weight Modifier component disabled
[SUCCESS] Success: Entity has a PostFX Layer component
[SUCCESS] Success: PostFX Gradient Weight Modifier component enabled
[SUCCESS] Success: Gradient Entity Id property set
[SUCCESS] Success: Opacity property set
[SUCCESS] Success: Invert Input property set
[SUCCESS] Success: Invert Input property set
[SUCCESS] Success: Enable Levels property set
[SUCCESS] Success: Input Max property set
[SUCCESS] Success: Input Min property set
[SUCCESS] Success: Input Mid property set
[SUCCESS] Success: Output Max property set
[SUCCESS] Success: Output Min property set
[SUCCESS] Success: Enable Transform property set
[SUCCESS] Success: Scale property set
[SUCCESS] Success: Rotate property set
[SUCCESS] Success: Translate property set
[SUCCESS] Success: Entered game mode
[SUCCESS] Success: Exited game mode
[SUCCESS] Success: Entity is hidden
[SUCCESS] Success: Entity is visible
[SUCCESS] Success: Entity deleted
[SUCCESS] Success: UNDO deletion success
[SUCCESS] Success: REDO deletion success
Test result:  SUCCESS
```
```

.\scripts\ctest\ctest_entrypoint.cmd --build-path atom_dev --suite main --ctest-executable "C:\Program Files\CMake\bin\ctest.exe" --config profile --tests-regex "AutomatedTesting::Atom_Main_Null_Render" --generate-xml
Starting 'main' suite: The default set of tests, covers most of all testing.
Executing CTest None time(s) with command:
  C:\Program Files\CMake\bin\ctest.exe --build-config profile --output-on-failure --parallel 16 --no-tests=error --label-exclude ^(SUITE_smoke|SUITE_periodic|SUITE_benchmark|SUITE_sandbox|SUITE_awsi)$ -T Test --tests-regex AutomatedTesting::Atom_Main_Null_Render
in working directory:
  atom_dev

    Start 164: AutomatedTesting::Atom_Main_Null_Render_03.main::TEST_RUN
1/4 Test #164: AutomatedTesting::Atom_Main_Null_Render_03.main::TEST_RUN ...   Passed  138.38 sec
    Start 161: AutomatedTesting::Atom_Main_Null_Render_00.main::TEST_RUN
2/4 Test #161: AutomatedTesting::Atom_Main_Null_Render_00.main::TEST_RUN ...   Passed  144.51 sec
    Start 162: AutomatedTesting::Atom_Main_Null_Render_01.main::TEST_RUN
3/4 Test #162: AutomatedTesting::Atom_Main_Null_Render_01.main::TEST_RUN ...   Passed  139.28 sec
    Start 163: AutomatedTesting::Atom_Main_Null_Render_02.main::TEST_RUN
4/4 Test #163: AutomatedTesting::Atom_Main_Null_Render_02.main::TEST_RUN ...   Passed  154.05 sec

100% tests passed, 0 tests failed out of 4

Label Time Summary:
COMPONENT_Atom      = 576.22 sec*proc (4 tests)
FRAMEWORK_pytest    = 576.22 sec*proc (4 tests)
SUITE_main          = 576.22 sec*proc (4 tests)

Total Test time (real) = 576.45 sec
```
some local asset processor issues slowed the results

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/Atom/atom_utils/atom_constants.py [hunk 1]
```diff
@@ -1003,13 +1025,39 @@ def postfx_gradient(property: str = 'name') -> str:
         """
         PostFX Gradient Weight Modifier component properties. Requires PostFX Layer component.
           - 'requires' a list of component names as strings required by this component.
-            Use editor_entity_utils EditorEntity.add_components(list) to add this list of requirements.\n
+            Use editor_entity_utils EditorEntity.add_components(list) to add this list of requirements.
+          - 'Gradient Entity Id' a separate entity id containing a gradient component.
+          - 'Opacity' factor multiplied by the current gradient before mixing. (float 0.0 to 1.0)
+          - 'Invert Input' swap the gradient input order black/white behave oppositely (bool)
+          - 'Enable Levels' toggle the application of input/output levels (bool)
```
Comment (NeilGitGud) [inline AutomatedTesting/Gem/PythonTests/Atom/atom_utils/atom_constants.py:1032]: Input and output comment lines (5 lines below this) exceed line length guidelines

File: AutomatedTesting/Gem/PythonTests/Atom/atom_utils/atom_constants.py [hunk 2]
```diff
@@ -990,11 +1002,21 @@ def physical_sky(property: str = 'name') -> str:
     def postfx_layer(property: str = 'name') -> str:
         """
         PostFX Layer component properties.
```
Comment (NeilGitGud) [inline AutomatedTesting/Gem/PythonTests/Atom/atom_utils/atom_constants.py:1004]: line 1005 exceeds line length guidelines

File: AutomatedTesting/Gem/PythonTests/Atom/atom_utils/atom_constants.py [hunk 3]
```diff
@@ -990,11 +1002,21 @@ def physical_sky(property: str = 'name') -> str:
...
+          - 'Layer Category' frequency at which the settings will be applied from atom_constants.py POSTFX_LAYER_CATEGORY
+          - 'Priority' this will take over other settings with the same frequency. lower takes precedence (int)
+          - 'Weight' how much these settings override previous settings. (float 0.0 to default 1.0)
```
Comment (NeilGitGud) [inline AutomatedTesting/Gem/PythonTests/Atom/atom_utils/atom_constants.py:1007]: line 1008, and 1009 exceeds line length guidelines

File: AutomatedTesting/Gem/PythonTests/Atom/atom_utils/atom_constants.py [hunk 4]
```diff
@@ -1033,8 +1081,9 @@ def postfx_shape(property: str = 'name') -> str:
         """
         PostFX Shape Weight Modifier component properties. Requires PostFX Layer and one of 'shapes' listed.
           - 'requires' a list of component names as strings required by this component.
-            Use editor_entity_utils EditorEntity.add_components(list) to add this list of requirements.\n
+            Use editor_entity_utils EditorEntity.add_components(list) to add this list of requirements.
```
Comment (NeilGitGud) [inline AutomatedTesting/Gem/PythonTests/Atom/atom_utils/atom_constants.py:1084]: Line 1085 barely exceeds line length guidelines

**Review by NeilGitGud — CHANGES_REQUESTED** [review]

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_PostFXGradientWeightModifierAdded.py [hunk 5]
```diff
@@ -66,24 +102,37 @@ def AtomEditorComponents_PostFXGradientWeightModifier_AddedToEntity():
     Test Steps:
     1) Create a PostFX Gradient Weight Modifier entity with no components.
     2) Add a PostFX Gradient Weight Modifier component to PostFX Gradient Weight Modifier entity.
-    3) UNDO the entity creation and component addition.
-    4) REDO the entity creation and component addition.
+    3) Remove PostFX Gradient Weight Modifier component.
+    4) UNDO component removal.
     5) Verify PostFX Gradient Weight Modifier component not enabled.
     6) Add PostFX Layer component since it is required by the PostFX Gradient Weight Modifier component.
```
[the remark this answers is not in the document]
Comment (NeilGitGud) [reply AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_PostFXGradientWeightModifierAdded.py:108]: fixed

**Review by NeilGitGud — APPROVED** [review]

**Review by galibzon — APPROVED** [review]

**Pull request merged**